grants/search.js

```diff
@@ -84,18 +94,84 @@ async function buildMatchCriteria(queryParams) {
     return match;
 }
 
+function buildFacetCriteria() {
+    return {
+        amountAwarded: [
+            {
+                $bucket: {
+                    groupBy: '$amountAwarded',
+                    boundaries: [0, 10000, 100000, 1000000, Infinity],
+                    output: {
+                        count: { $sum: 1 }
+                    }
+                }
+            }
+        ],
+        awardDate: [
+            {
+                $group: {
+                    _id: {
+                        $year: '$awardDate'
+                    },
+                    count: { $sum: 1 }
+                }
+            }
+        ],
+
+        grantProgramme: [
+            {
+                $group: {
+                    _id: {
+                        $arrayElemAt: ['$grantProgramme.title', 0]
+                    },
+                    count: { $sum: 1 }
+                }
+            },
+            { $sort: { _id: 1 } }
+        ],
+
+        orgType: [
+            {
+                $group: {
+                    _id: {
+                        $arrayElemAt: [
+                            '$recipientOrganization.organisationType',
+                            0
+                        ]
+                    },
+                    count: { $sum: 1 }
+                }
+            },
+            { $sort: { _id: 1 } }
+        ]
+    };
+}
+
 async function fetchGrants(collection, queryParams) {
     const perPageCount =
         (queryParams.limit && parseInt(queryParams.limit)) || 50;
     const pageParam = queryParams.page && parseInt(queryParams.page);
     const currentPage = pageParam > 1 ? pageParam : 1;
     const skipCount = perPageCount * (currentPage - 1);
 
+    const matchCriteria = await buildMatchCriteria(queryParams);
+    const facetCriteria = buildFacetCriteria();
+
     /**
      * Construct the aggregation pipeline
      */
-    const matchCriteria = await buildMatchCriteria(queryParams);
-    const pipeline = [{ $match: matchCriteria }];
+    const resultsPipeline = [
+        { $match: matchCriteria },
+        { $sort: { awardDate: -1 } },
+        {
+            $addFields: {
+                id: {
+                    // Strip 360Giving prefix from returned ID
+                    $arrayElemAt: [{ $split: ['$id', '360G-blf-'] }, 1]
```

Comment:
Maybe make this a shared constant somewhere (you never know...) for the other search/replace?